engines/optix/OptiXModel.h

```diff
@@ -114,6 +121,19 @@ class OptiXModel : public Model
     std::map<std::string, optix::Buffer> _optixTextures;
     std::map<std::string, optix::TextureSampler> _optixTextureSamplers;
 
+    // Transfer function
+    struct OptiXTransferFunction
+    {
+        optix::Buffer colors{nullptr};         // vec3fs
+        optix::Buffer opacities{nullptr};      // floats
+        optix::Buffer simulationData{nullptr}; // floats
```

Comment:
This is semantically incorrect. The simulation data is not part of the transfer function, but the input to the function. Move it out from here rename commmitTransferFunction to something else, for example commitDataMapping or whatever. If you can't find a good name make _commitSimulationData and _commitTransferFunction public instead.